tests/test_abstract_pair_converter.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2023 Neal Chambers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import DefaultDict
+
+from prezzemolo.graph import Graph
+from prezzemolo.vertex import Vertex
+
+from dali.abstract_pair_converter_plugin import GraphVertexesDict
+
+
+class TestAbstractPairConverterPlugin:
+    def test_graph_vertexes_class(self) -> None:
+        current_graph: Graph[str] = Graph[str]()
+        graph_vertexes: DefaultDict[str, Vertex[str]] = GraphVertexesDict(current_graph)
+
+        first_vertex = graph_vertexes["first vertex"]
```

Comment:
This approach is clever but I'm afraid it makes the code less readable, because you redefined a common behavior of dictionaries: if `graph_vertexes` was a plain dict, then `graph_vertexes["first vertex"]` would have no side effects. But the `GraphVertexesDict` adds side effects under the hood that would change the default semantics of dict in a way makes it more difficult to follow the logic for somebody new to the code (and I argue, even for somebody who's familiar with the semantics of `GraphVertexesDict`, but who hasn't worked on the code for a while). This would throw off most Python programmers: it is a bit like operator redefinition, which is fine in certain limited circumstances, but it can lead to trouble understanding and maintaining the code down the road. 

I would prefer an approach that makes the operations explicit and clear to the reader, without redefining standard behavior. Here's a potential approach based on inheritance, that doesn't modify standard semantics:
```
class MappedGraph (Graph(Generic[ValueType])):
    def __init__(self, vertexes: Optional[List["Vertex[ValueType]"]] = None) -> None:
        super().__init__(vertexes)
        self.__name_to_vertex: Dict[str, Vertex[ValueType]] = {vertex.name:vertex for vertex in vertexes}

    def get_vertex(self, name:str) -> Vertex[ValueType]:
        return self.__name_to_vertex[name]

    def add_neighbor(self, vertex: "Vertex[ValueType]", weight: float = 0.0) -> None:
        super().add_neighbor(vertex, weight)
        self.__name_to_vertex[vertex.name] = vertex
```

Another approach would be to use composition, but probably the inheritance-based one is more compact.

Reply:
> And then add a get_or_set method for creating new vertexes:
> 
> ```
> def get_or_set_vertex(self, name:str) -> Vertex[ValueType]:
>       if self.get_vertex(name):
>             return self.get_vertex(name)
>       else:
>             new_vertex: Vertex[ValueType] = Vertex[str](name=name)
>             super().add_vertex(new_vertex)
>             self.__name_to_vertex[name] = new_vertex
>             return new_vertex
> ```

Sounds good.

> And shouldn't `get_vertex` be:
> 
> ```
> def get_vertex(self, name:str) -> Optional[Vertex[ValueType]]:
>         return self.__name_to_vertex.get(name)
> ```
> 
> So that it mimics `get()` behavior?

Looks good, too.

> And, the `add_neighbor()` would be:
> 
> ```
>     def add_neighbor(self, vertex_name: str, neighbor_name: str, weight: float = 0.0) -> None:
>         vertex: Vertex[str] = self.get_or_set_vertex(vertex_name)
>         neighbor: Vertex[str] = self.get_or_set_vertex(neighbor_name)
>         if neighbor not in vertex.neighbors:
>             vertex.add_neighbor(neighbor, weight)
> ```

Looks great, but be change the neighbor check to:
```
         if not vertex.has_neighbor(neighbor):
             vertex.add_neighbor(neighbor, weight)
```
Because it's hashtable-based search rather iterator-based (so it's much faster).

> This is excellent since it eliminates the need for two `dict`s. We just need one that tracks graphs and vertexes together.

Awesome!

Reply:
This will most likely grow in size since I might as well add `markets` (ie. edges) to this as well. I'd like to implement a priorityqueue that prioritizes markets from exchanges with higher volume. However, that is out of scope for this PR. 

Should I separate it out now into its own file?

Reply:
Do you mean size as is lines of code? You're thinking of separating MappedGraph into a separate class?